### PR TITLE
Vmbackup: Provide client-side TLS configuration to create/delete snapshot

### PR DESCRIPTION
**Pull request by khushijain21**
Completes #5724

### CODE REVIEW

File: lib/snapshot/snapshot.go [hunk 1]
```diff
@@ -12,10 +13,18 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/utils"
```
Comment (zekker6) [inline lib/snapshot/snapshot.go:16]: Imports from `lib/..` packages should not import packages from `app/...` packages as the idea is to do this another way around. All re-usable code should be a part of `lib`.
Could you please move `app/vmalert/utils/tls.go` to `lib/httputils` and update references to it here and in `app/vmalert` to point to a new location?

File: lib/snapshot/snapshot.go [hunk 2]
```diff
@@ -12,10 +13,18 @@ import (
...
 	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
 )
 
-var snapshotNameRegexp = regexp.MustCompile(`^[0-9]{14}-[0-9A-Fa-f]+$`)
+var (
+	snapshotNameRegexp    = regexp.MustCompile(`^[0-9]{14}-[0-9A-Fa-f]+$`)
+	tlsInsecureSkipVerify = flag.Bool("snapshot.tlsInsecureSkipVerify", false, "Whether to skip tls verification when connecting to -snapshotCreateURL")
```
Comment (zekker6) [inline lib/snapshot/snapshot.go:22]: Could you please update https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/docs/vmalert.md#flags section of documentation with the new flags added?

**Review by zekker6 — CHANGES_REQUESTED** [review]
Hello @khushijain21, thank you for the pull request!

Please, take a look at a few comments. Also, it would be great to add an entry about this into a changelog [here](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/docs/CHANGELOG.md).

File: lib/snapshot/snapshot.go (hunk 2 above)
Comment (khushijain21) [reply lib/snapshot/snapshot.go:22]: did you mean vmbackup documentation instead?

Comment (zekker6) [reply lib/snapshot/snapshot.go:22]: Yes, that's correct. Sorry for confusion :)

**Review by zekker6 — APPROVED** [review]
LGTM, thanks!

File: lib/snapshot/snapshot.go [hunk 3]
```diff
@@ -30,7 +39,15 @@ func Create(createSnapshotURL string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	resp, err := http.Get(u.String())
+
+	// create Transport
+	tr, err := httputils.Transport(createSnapshotURL, *tlsCertFile, *tlsKeyFile, *tlsCAFile, *tlsServerName, *tlsInsecureSkipVerify)
+	if err != nil {
+		logger.Fatalf("failed to create transport: %s", err)
```
Comment (Haleygo) [inline lib/snapshot/snapshot.go:46]: I think we should return error msg here like we did below, let the caller to log with more context. 
```
	resp, err := hc.Get(u.String())
	if err != nil {
		return "", err
	}
```

File: lib/snapshot/snapshot.go [hunk 4]
```diff
@@ -12,10 +13,18 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/httputils"
 	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
 )
 
-var snapshotNameRegexp = regexp.MustCompile(`^[0-9]{14}-[0-9A-Fa-f]+$`)
+var (
+	snapshotNameRegexp    = regexp.MustCompile(`^[0-9]{14}-[0-9A-Fa-f]+$`)
```
Comment (Haleygo) [inline lib/snapshot/snapshot.go:21]: I think this variable can be defined apart from flag vars, wdyt?

File: lib/snapshot/snapshot.go [hunk 5]
```diff
@@ -68,7 +85,13 @@ func Delete(deleteSnapshotURL string, snapshotName string) error {
 	if err != nil {
 		return err
 	}
-	resp, err := http.PostForm(u.String(), formData)
+	// create Transport
+	tr, err := httputils.Transport(deleteSnapshotURL, *tlsCertFile, *tlsKeyFile, *tlsCAFile, *tlsServerName, *tlsInsecureSkipVerify)
+	if err != nil {
+		logger.Fatalf("failed to create transport: %s", err)
```
Comment (Haleygo) [inline lib/snapshot/snapshot.go:91]: Same as above comment, return the error here.

**Review by valyala — APPROVED** [review]

File: docs/CHANGELOG.md [hunk 6]
```diff
@@ -55,8 +55,11 @@ The v1.97.x line will be supported for at least 12 months since [v1.97.0](https:
 * BUGFIX: [vmagent](https://docs.victoriametrics.com/vmagent.html): fix the increased CPU usage when sending the data to remote storage. The issue has been introduced in [v1.97.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.97.0).
 * BUGFIX: fix `runtime error: slice bounds out of range` panic, which can occur during query execution. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5733). The bug has been introduced in `v1.97.0`.
 * BUGFIX: [MetricsQL](https://docs.victoriametrics.com/MetricsQL.html): properly handle `avg_over_time({some_filter}[d]) keep_metric_names` queries, where [`some_filter`](https://docs.victoriametrics.com/keyconcepts/#filtering) matches multiple time series with multiple names, while `d` is bigger or equal to `3h`. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5556).
+* BUGFIX: dashboards/single: fix typo in query for `version` annotation which falsely produced many version change events.
```
Comment (Haleygo) [inline docs/CHANGELOG.md:58]: Hey, seems there was a rephrase for this item and this is replaced by line60, so not needed.
>* BUGFIX: [dashboards/single](https://grafana.com/grafana/dashboards/10229): fix typo in query for `version` annotation which falsely produced many version change events.

**Review by Haleygo — APPROVED** [review]
LGTM!

**Review by hagen1778 — APPROVED** [review]
LGTM!

**Comment by valyala** [comment]
FYI, this pull request has been included in [v1.98.0](https://docs.victoriametrics.com/CHANGELOG.html).

**Pull request merged**